src/appleseed/renderer/kernel/lighting/pt/ptlightingengine.cpp

```diff
@@ -1113,10 +1110,13 @@ namespace
                     }
 
                     radiance *= vertex.m_throughput;
-                    radiance *= transmission;
+
                     const float scalar_weight = current_mis_weight /
                         (distance_sample_count_float * distance_prob);
-                    madd(m_path_radiance, radiance, scalar_weight);
+                    radiance *= transmission * scalar_weight;
```

Comment:
This creates a `Spectrum` temporary which is easily avoided (also getting rid of `scalar_weight`):

    radiance *= transmission;
    radiance *= current_mis_weight / (distance_sample_count_float * distance_prob);